src/sss_client/nss_mc_common.c

```diff
@@ -77,6 +98,11 @@ errno_t sss_nss_check_header(struct sss_cli_mc_ctx *ctx)
     int ret;
     struct stat fdstat;
 
+    ret = sss_nss_mc_validate(ctx);
```

Comment:
There is:
```
    ret = fstat(ctx->fd, &fdstat);
    if (fdstat.st_nlink == 0) {
        /* memory cache was removed; we need to reinitialize it. */
        return EINVAL;
    }
```
below in this functions.

Is this duplication needed?

Reply:
I don't know. We need to check it before MEMCPY_WITH_BARRIERS to avoid the crash. I don't know why the other check wasn't put before MEMCPY_WITH_BARRIERS in the first place, so perhaps there is some use for it?

Reply:
Probably order of checks wasn't so important with link counter...

I would include `(fdstat.st_nlink == 0)` check into `sss_nss_mc_validate()` and remove 2nd `fstat()` to avoid perf hit.

Reply:
Ok, done.